coalib/bearlib/aspects/decorators.py

```diff
@@ -49,3 +49,53 @@ def _new_func(self, *args, **kwargs):
         return _new_func
 
     return _func_decorator
+
+
+def map_ambiguous_setting_to_aspect(**aspectable_setting):
+    """
+    Convert bear settings to use aspects in which there is some ambiguity.
+    The ambiguity is because of differing values (maybe either name or even
+    data-types) in bear settings and their corresponding aspect tastes.
+
+    This decorator is used by ``Bear.run()`` to automatically map and
+    override the value in the bear setting with their equivalent taste
+    as provided in the parameter.
+
+    The order of setting override from the highest to lowest is:
+    - Explicit setting
+    - Explicit aspect/taste default (if aspect is activated in Section)
+    - Aspect/taste default (if aspect is activated in Section)
+    - Setting default (in bear's run argument)
+
+    :param aspectable_setting:
+        A dictionary in which keys are the bear settings and the corresponding
+        value for each key is a tuple containing an equivalent taste as first
+        value and a list of tuples with matching bear settings' and aspects'
+        value as the second value.
+    """
+    def _func_decorator(func):
+        @wraps(func)
+        def _new_func(self, *args, **kwargs):
+            if self.section.aspects:
```

Comment:
Always bail out early instead of using indentation which makes the code unreadable as more conditions are added.

```py
if not self.section.aspects:
    return

...
```

Reply:
I have to do `return func(self, *args, **kwargs)` instead of `return` so that if aspects aren't there, the already existing default bear settings can be applied, otherwise it will return me an empty dict :)